.changes/unreleased/Added-20250603-162638.yaml

```diff
@@ -0,0 +1,42 @@
+kind: Added
+body: |-
+  Unbundle the SDK interface to support partial implementation.
+
+  When you implement a custom SDK, you can now choose to just implement part of the `SDK` interface depending on your needs.
+  That means:
+  - `dagger develop` no longer fails if your SDK just implement the `Runtime` interface, it will simply not call `Codegen` if not defined.
+  - `dagger call` and `dagger functions` give a clear error message if it's not supported by the SDK.
```

Comment:
I don't think the changelog needs to be this long - we can just say that partial SDKs are allowed, and explain what that means. 

For a full example, users can look at this PR - it's not a normal user feature, it's definitely for power-users.